# Overview
This repository contains files as in response to the selection test for the project "Creating a new pipeline for long-read RNA-seq using StringTie v2" of C3G given [here](https://bitbucket.org/mugqic/gsoc_2020/src/master/) 

## Files
The repository contains various files, each one is explained below- <br/> <br/>
**1) ST2_TM_cmds.sh** - This bash file contains the commands for StringTie2 & Tablemaker <br/>
It uses long read mode of StringTie on the given bam file to generate the .gtf file, (-L for long-reads, -p 6 for increasing the no. of threads, -v for verbose output) <br/> 
`stringtie -L RNA_nanopore.brain.C1R1_mapping_E94_minimap2_primary_no_read_less_than_80QC.bam -o RNA_nanopore.brain.gtf -p 6 -v` <br/>
//ly for liver file <br/>
Next, it uses tablemaker command to generate the .ctab files for ballgown to import (again -p 6 for multithreading, also it uses fr-firststrand as its library-type ) <br/>
`tablemaker -p 4 -q -W -G RNA_nanopore.brain.gtf -o output_brain RNA_nanopore.brain.C1R1_mapping_E94_minimap2_primary_no_read_less_than_80QC.bam  --library-type fr-firststrand` <br/> <br/>
**2) RNA_nanopore.brain_1000.gtf** - Contains the 1st 1000 lines output generated from StringTie, <br/> `head -1002 xxx.gtf > xxx_1000.gtf` has been used to filter the same. <br/> //ly for **RNA_nanopore.liver_1000.gtf** <br/> <br/>
**3) ball_input** - This folder contains 2 sub folders, each one containing ctab files generated by the tablemaker command & to be used by ballgown. <br/> <br/>
**4) ball_cmds.R** - This is the R file for importing ouput of Tablemaker using ballgown, code has been explained in the file itself. <br/> <br/>
**5) tx_brain_1000.csv** - It contains the ouptut generated using texpr function in Ballgown. Again 1st 1000 lines has been filtered using `head -1001 xxx.csv > xxx_1000.csv`


